pkg/lease/manager.go

```diff
@@ -227,6 +227,12 @@ func (l *manager) invalidateLease(ctx context.Context, obj client.Object) error
 		}
 		return err
 	}
+	if l.holderIdentity != *lease.Spec.HolderIdentity {
+		isLeaseExpired := time.Now().After(lease.Spec.RenewTime.Time.Add(time.Duration(*lease.Spec.LeaseDurationSeconds) * time.Second))
```

Comment:
I don't think we should invalidate not owned leases, even when they expired.

(and nil checks are missing, and we should reuse isValidLease, but that doesn't matter when we remove this...)

Reply:
>I don't think we should invalidate not owned leases, even when they expired.

why not ? IIRC RequestLease would work for this use case

Reply:
- what's the use case for *invalidating* not owned leases? 🤷🏼‍♂️ 
 - why adding more code than needed for fixing this bug after code freeze?

Reply:
>what's the use case for invalidating not owned leases? 🤷🏼‍♂️

I don't have a particular use case in mind, however I do think it makes sense to align the behavior to that of RequestLease, I think it's quite confusing that we will not allow deleting an expired lease directly (through `Invalidate` API)  but it is fine to do it if we use `RequestLease` API.

>why adding more code than needed for fixing this bug after code freeze?

I do agree that we should minimize code inserted this late before the release, however in this case it's a single "if" statement so  IMO the value outweigh the risk.
I think that a better approach to follow this principal would be having the fix only at NHC instead of both repos.

Reply:
>  align the behavior to that of RequestLease

There we have clear usecase.

> I think it's quite confusing that we will not allow deleting an expired lease

I think it's more confusing to delete leases you don't own. When I create a lease and it's deleted by someone else, at least I would be surprised.

> however in this case it's a single "if" statement

which had two missing nil checks in its 1st iteration...

>  a better approach to follow this principal would be ATM having a fix only at NHC

which would need more code, and would leave the issue here  for the next one...

Reply:
ATM looks like we don't agree on whether `InvalidateLease` should only work on owned leases or not.
In order to get on with the release I've made the change so only owned leases will be Invalidated and we can have further discussion later on.